Utils/Dataflow/060_upload2virtuoso/uploadTTL.sh

```diff
@@ -231,8 +250,23 @@ done
 [ -z "$HOST" ] && echo "(ERROR) empty host value." >&2 && exit 2
 [ -z "$PORT" ] && echo "(ERROR) empty port value." >&2 && exit 2
 [ -z "$GRAPH" ] && GRAPH=http://$HOST:$PORT/$GRAPH_PATH
-[ -z "$DELIMITER" ]  && DELIMITER=$'\0'
-[ "x$DELIMITER" = "xNOT SPECIFIED" ] && DELIMITER=$'\n'
+
+[ -n "$EOB" ] && EOBatch="$EOB"
+[ -n "$EOP" ] && EOProcess="$EOP"
+
+case $BATCHMODE in
+  e|enabled)
+    [ -z "$EOB" ] && EOBatch="\n"
+    ;;
+  d|disabled)
+    ( [ -z "$EOB" ] || [ $EOB == "\x17" ] || [ $EOBatch == "\x17" ] ) && EOBatch="\n"
+    ( [ -n "$EOB" ] && [ ! $EOB == "\x17" ] ) && EOBatch="$EOB"
```

Comment:
Not sure I\`m getting this `case` statement properly... It is similar to the one puzzled me in `069_upload2es/load_data.sh`, but here we have slightly different initial conditions...

First, what does it say now (suggesting, that `EOB` should be used as `EOP`, for custom value passed with `--eob`):
"If batch mode enabled, EOBatch is `\n` unless EOB is defined. If batch mode is disabled, EOBatch is `\n` unless EOB is defined with anything but `\x17` -- else EOBatch is EOB." 

Concerning conditions... as we have this lines above:
```
...
    -B|--eob)
      EOB=`echo -ne $2` # allowed myself to replace EOBatch with EOB
...
[ -n "$EOB" ] && EOBatch="$EOB"
```
within the `case`:
* if `EOB` is empty, `EOBatch` should be `"\x17"` (the default value);
* if `EOB` is `"\x17"` (or, better to say, if user passed `--eob \x17` in the command line), then `EOB` is not *string* `"\x17"`, but ASCII code ETB;
* if, after all, `EOB` *is* string `"\x17"`, then `EOBatch` is `"\x17"` too.

So, in plain words -- what should be done during this `case`?

Reply:
@mgolosova 

I supposed that `case` allows here to check the following conditions:

if batch mode is disabled (by hands or by default):
1. no EOB = `-b ''` -->  `$EOB='\n'`
2. we have EOB, but \x17 (hex) --> `$EOB='\n'` (it might be unnecessary and too strict, but I'm not sure)
3. we have any other EOB --> `$EOB='EOB'`

if batch mode is enabled: 
1. only if we type `--eob ''` --> `$EOB='\n'`

Reply:
Alright. I think there\`s a little bit of mess happened with enabled/disabled batch mode etc...

---

First, about this (just to make sure we do understand things same way).
@anastasiakaida wrote:
> we have EOB, but \x17 (hex) --> $EOB='\n' (it might be unnecessary and too strict, but I'm not sure)

This checks:
```
[ ! $EOB == "\x17" ]
[ $EOB == "\x17" ]
```
are string comparison. Works this way:
```
mgolosova@artois-K:~$ a_str="\x17"
mgolosova@artois-K:~$ a_ascii=$'\x17'
mgolosova@artois-K:~$ echo "str: '$a_str'; ASCII: '$a_ascii'"
str: '\x17'; ASCII: ''
mgolosova@artois-K:~$ [ "$a_str" == "\x17" ] && echo "True" || echo "False"
True
mgolosova@artois-K:~$ [ "$a_ascii" == "\x17" ] && echo "True" || echo "False"
False
 ```
Now `EOB` is assigned with string value, so it\`s OK; just wanted to make things clear.

---

Next, about the general logic.
I have tested it with attached script: [test_batchmode.txt](https://github.com/PanDAWMS/dkb/files/2289531/test_batchmode.txt)

Here\`s the results:
```
   -b   ||  X   |  X   |   X    |   X    || 'e'  | 'e'  |  'e'   |  'e'   |
------- || ---- | ---- | ------ | ------ || ---- | ---- | ------ | ------ | 
   -B   ||  X   |  ''  | '\x17' | '\x11' ||  X   |  ''  | '\x17' | '\x11' |
======= || ==== | ==== | ====== | ====== || ==== | ==== | ====== | ====== |
EOBatch || '\n' | '\n' |  '\n'  | '\x11' || '\n' | '\n' | '\x17' | '\x11' |
```

How do I read it: batch mode is actually enabled in three cases:
* `-B 'non-default marker'`;
* `-b -B 'default marker'`;
* `-b -B 'non-default marker'`.

While I would expect the following table:
```
   -b   ||  X   |  X   |   X    |   X    ||  'e'   | 'e'  |  'e'   |  'e'   |
------- || ---- | ---- | ------ | ------ || ------ | ---- | ------ | ------ | 
   -B   ||  X   |  ''  | '\x17' | '\x11' ||   X    |  ''  | '\x17' | '\x11' |
======= || ==== | ==== | ====== | ====== || ====== | ==== | ====== | ====== |
EOBatch || '\n' | '\n' | '\x17' | '\x11' || '\x17' | '\n' | '\x17' | '\x11' |
```
Two cases added:
* `-b`;
* `-B 'default marker'`.

In other words:
* it shouldn\`t matter if user passed default marker or not: if it is passed implicitly, it is preferable value;
* if user said "enable batch mode", then we should use default EOB for batch mode.

Reply:
*) And about forgotten quotes: it was totally OK; shell takes non-quoted stuff as a string unless it is a keyword or the string is followed by `=` (meaning that it is variable assignment). Quotes are required for multi-word strings (though they could be written without quotes too, if spaces are escaped) and to escape special characters.
See:
```
mgolosova@artois-K:~$ a=abcdef
mgolosova@artois-K:~$ echo $a
abcdef
mgolosova@artois-K:~$ echo abcdef
abcdef
mgolosova@artois-K:~$ a=abc def
def: command not found
mgolosova@artois-K:~$ a=abc\ def
mgolosova@artois-K:~$ echo $a
abc def
mgolosova@artois-K:~$ echo abc\def
abcdef
mgolosova@artois-K:~$ echo abc\ def
abc def
mgolosova@artois-K:~$ echo "abc\ def"
abc\ def
mgolosova@artois-K:~$ echo "abc def"
abc def
```

So in `case` it is OK to write strings without quotes (as values, not keywords or something are expected); moreover, if a wildcard '*' is used, quotes will escape it and it will be taken as an asterisk character, not a wildcard:
```
mgolosova@artois-K:~$ a=abcd
mgolosova@artois-K:~$ case $a in
> "a*")
>   echo 'Quoted case: "a*"';;
> a*)
>   echo 'Non quoted case: a*';;
> *)
>   ;;
> esac
Non quoted case: a*
```

Yet in this case quotes are also OK to use, as no wildcard or something is used.